### PR TITLE
Expose as vendor module

### PR DESCRIPTION
**Pull request by chillu**
See https://github.com/silverstripe/silverstripe-framework/issues/7405
Merge with https://github.com/silverstripe/silverstripe-installer/pull/200

 * [x] .htaccess in installer
 * [x] web.config in installer
 * [x] Installer with custom `_register_database.php` hooks
 * [x] Installer styling (incl. `frameworkmissing.html`)
 * [x] Generate .htaccess for secure files
 * [x] Update existing .htaccess for secure files (forced through flush)
 * [x] TinyMCE generation in dev and live mode
 * [x] Operation through `php -S 127.0.0.1:80 index.php`
 * [x] Run behat with module as webroot (tested through Travis)
 * [x] Run behat with module in vendor
 * [x] Docs: Module directories
 * [x] Docs: Secure files
 * [x] Upgrading docs

Test (after merging the framework pull request):

```
composer config repositories.framework vcs https://github.com/open-sausages/silverstripe-framework.git
composer require "silverstripe/framework:dev-pulls/4/vendorise-me-baby as 4.0.x-dev"
```

### CODE REVIEW

File: src/Dev/Install/Installer.php [hunk 1]
```diff
@@ -440,7 +441,7 @@ public function createHtaccess()
 
     # Deny access to vendor, unless you're requesting main.php
     # Not restricting to the start of the path to support RewriteBase
-    RewriteCond %{REQUEST_URI} !/vendor/silverstripe/framework/main\.php
+    RewriteCond %{REQUEST_URI} !^/vendor/silverstripe/framework/main\.php
```
Comment (flamerohr) [inline src/Dev/Install/Installer.php]: what does the added `^` do?

Comment (tractorcow) [reply src/Dev/Install/Installer.php]: I just ensures vendor is at the start of the path.

Comment (chillu) [reply src/Dev/Install/Installer.php]: If anyone can point me to a way to make that work with `RewriteBase` I'm happy to add it back in :)

Comment (tractorcow) [reply src/Dev/Install/Installer.php]: I've got it working without rewritebase.

**Pull request merged**